packages/server/src/api/controllers/table/external.ts

```diff
@@ -82,15 +83,30 @@ export async function bulkImport(
   ctx: UserCtx<BulkImportRequest, BulkImportResponse>
 ) {
   let table = await sdk.tables.getTable(ctx.params.tableId)
-  const { rows } = ctx.request.body
+  const { rows, identifierFields } = ctx.request.body
   const schema = table.schema
 
+  if (
+    identifierFields &&
+    identifierFields.length > 0 &&
+    !_.isEqual(identifierFields, table.primary)
+  ) {
+    // This is becuse we make use of the ON CONFLICT functionality in SQL
+    // databases, which only triggers when there's a conflict against a unique
+    // index. The only unique index we can count on atm in Budibase is the
+    // primary key, so this functionality always uses the primary key.
+    ctx.throw(
+      400,
+      "Identifier fields are not supported for bulk import into an external datasource."
```

Comment:
Would it not be better here to give an exact error "Only primary keys can be used for bulk importing of rows"- something that should be obvious to the user.

Reply:
This error should, in theory, never be shown to the user because the UI doesn't let them specify the identifier fields. Happy to change it either way, but the idea is they'd never see it 😅